docs/installation/docker-compose.md

```diff
@@ -0,0 +1,299 @@
+
+
+2. Enter the project directory:
+
+        :::bash
+        cd autolab-docker
+
+3. Update our Autolab and Tango submodules to ensure that you are getting the latest versions:
+
+        :::bash
+        make update
+
+4. Create initial default configs:
+
+        :::bash
+        make
+
+5. Build the Dockerfiles for both Autolab and Tango:
+
+        :::bash
+        docker-compose build
+
+6. Run the Docker containers:
+
+        :::bash
+        docker-compose up -d
+
+    Note at this point Nginx will still be crash-looping in the Autolab container because SSL has not been configuired/disabled yet.
+
+7. Ensure that the newly created config files have the right permissions, as it may have been modified during the building process:
+
+        :::bash
+        make set-perms
+
+8. Perform database migrations for Autolab, which will initialize your database schema:
+
+        :::bash
+        make db-migrate
+
+9. Create administrative user for Autolab:
+
+        :::bash
+        make create-user
+
+    This user has full permissions on Autolab and will be able to create other users and designate other admins.
+
+10. Change `DOCKER_TANGO_HOST_VOLUME_PATH` in `docker-compose.yml` to be the absolute path to the Tango `volumes` directory, i.e `/<path-to-docker-compose-installation>/Tango/volumes`. This is so that Tango knows where to put the output files of its autograded jobs.
+
+        :::bash 
+        # in docker-compose.yml
+
+        # Modify the below to be the path to volumes on your host machine
+        - DOCKER_TANGO_HOST_VOLUME_PATH=/home/your-user/autolab-docker/Tango/volumes # example path
+
+11. Stop all containers, as we are going to setup/disable TLS:
+
+        :::bash 
+        docker-compose stop
+
+12. If you intend to use TLS later, in `nginx/app.conf`, change instances of `<REPLACE_WITH_YOUR_DOMAIN>` to your real domain name. Otherwise, if you are not using TLS, in `nginx/no-ssl-app.conf`, change `server_name` to your real domain name.
+
+13. Continue with TLS setup as outlined in the next section
+14. Build the autograding image(s) that you want to use in Tango (see [the docs](https://docs.autolabproject.com/tango-vmms/#docker-vmms-setup) for more information). For this setup we will stick to the default Ubuntu 18.04 autograding image: 
+
+        :::bash
+        docker build -t autograding_image Tango/vmms/
+        
+Note that we can just run this directly on the host because we are mapping the Docker socket to the Tango container (i.e they are using the same Docker server).
+
+15. Start up everything: 
+
+        :::bash
+        docker-compose up -d
+        
+Autolab should now be accessible on port 80 (and 443 if you configured TLS)!
+
+## Configuring SSL/TLS
+There are three options for TLS: using Let's Encrypt (for free TLS certificates), using your own certificate, and not using TLS (not recommended for production deployment).
```

Comment:
I think it would be good to explain a bit of SSL/TLS context. 

And I would suggest instead of saying not recommended for production deployment, could say "suitable for local testing / development"

Reply:
Could tell them a production deployment is for actual use with students and instructors, while testing is just if u want to quickly try out autolab

Reply:
Updated